server/routes/api/auth.ts

```diff
@@ -75,6 +76,28 @@ const validationError = (res: Response, err: Error) => {
   });
 };
 
+const performAuthentication = (
```

Comment:
nginx authenticate the user based on the `.htpasswd` file provided, but the user authenticated by nginx could not be present in the flood db

Reply:
That's right. However, JWT authentication later (via token/cookie) will reject the request when the user is not present in the database.

Reply:
correct me if I'm wrong, but the JWT authentication only check if the signature is valid and in case of `requireAdmin` perform a validation on the signature payload, but it doesn't validate the user in the token against the db: that's why you can bypass auth with a user in configuration.
in this case we have to perform the authentication for the nginx credentials.

do I miss anything?

Reply:
Flood still verify if the user exists even if the JWT token is generated by server secret. There is a test case for this:
https://github.com/jesec/flood/blob/49086175036edf3cec14d628976454e6a310f5c8/server/routes/api/auth.test.ts#L241-L254

Reply:
yes, I finally found it in the `passport.ts` file

anyway not performing the authentication creates a rather annoying experience in case we generate the token from http credentials for a user that's not in the flood db: the client will load the overview and then all the requests will return 401 because of the failed jwt validation. so I'd prefer to perform the authentication, what do you think?

Reply:
No we should not do password authentication here. Authentication is supposed to be handled externally in this case. We don't need duplicate authentication. 

If you don't want the overview to be loaded, you have to change the "/verify" endpoint to return an error response when necessary.

Reply:
the only way to know if the http credentials are valid is looking for the user in the db, that's what `performAuthentication` does. I can change the name, but please look at the implementation.

- `/verify`
old flow didn't change, new flow compare with db and return `AuthVerificationResponse` instead of executing `passport.authenticate()`, exactly like the `disableUsersAndAuth` behaviour

- `/api/auth/authenticate`
didn't change anything, I only check if reading the credentials from body or `authorization` header